postgres/rdspostgres/rdspostgres.go

```diff
@@ -33,31 +32,12 @@ import (
 	"gocloud.dev/postgres"
 )
 
-// Params specifies how to connect to an RDS database.
-type Params struct {
-	// Endpoint is the host/port of the RDS database, like
-	// "myinstance.borkxyzzy.us-west-1.rds.amazonaws.com:5432".
-	// If no port is given, then 5432 is assumed.
-	Endpoint string
-
-	// User is the database user to connect as.
-	User string
-	// Password is the database user password to use.
-	Password string
-	// Database is the PostgreSQL database name to connect to.
-	Database string
-
-	// Values sets additional parameters, as documented in
-	// https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-PARAMKEYWORDS.
-	Values url.Values
-
-	// TraceOpts contains options for OpenCensus.
-	TraceOpts []ocsql.TraceOption
-}
-
 // URLOpener opens RDS PostgreSQL URLs
 // like "rdspostgres://user:password@myinstance.borkxyzzy.us-west-1.rds.amazonaws.com:5432/mydb".
 type URLOpener struct {
+	// CertSource specifies how the opener will obtain the RDS Certificate
+	// Authority. If nil, it will use the default *rds.CertFetcher.
+	CertSource rds.CertPoolProvider
```

Comment:
Why not use the "lazy" pattern used elsewhere here? Is getting the default CertFetcher just super cheap for some reason, while for `cloudpostgres` it is not?

Reply:
Yes. `rds.CertFetcher` does not require credentials (which is expensive to fetch), it purely fetches a root CA and can do that on-demand.